forced_photometry/code_src/galex_functions.py

```diff
@@ -0,0 +1,178 @@
+import os
+import re
+import shutil
+import urllib.request
+from pathlib import Path
+
+import numpy as np
+from astropy.coordinates import SkyCoord
+from astropy.table import Table, vstack
+import astropy.units as u
+import pyvo
+from astroquery.mast import Observations
+
+def galex_get_images(coords, search_radius_arcsec=60, min_exptime=40000, 
+                     output_dir="data/Galex", max_products_per_source=None, verbose=True):
+    """
+    Download GALEX science and sky background images for a given coordinate using astroquery.
+
+    Parameters
+    ----------
+    coords : astropy.coordinates.SkyCoord
+        Sky coordinates of the target region.
+    search_radius_arcsec : float, optional
+        Search radius in arcseconds. Default is 60.
+    min_exptime : float, optional
+        Minimum exposure time in seconds. Default is 40000.
+    output_dir : str, optional
+        Directory to store downloaded files. Default is "data/Galex".
+    max_products_per_source : int or None, optional
+        Maximum number of products to download. If None, download all. Default is None.
+    verbose : bool, optional
+        Whether to print status messages. Default is True.
+
+    Returns
+    -------
+    downloaded_files : list of str
+        List of paths to the downloaded files.
+    """
+
+    if verbose:
+        print(f"Querying GALEX around RA={coords.ra.deg:.5f}, Dec={coords.dec.deg:.5f}...")
+
+    Observations.enable_cloud_dataset()
+    obs_table = Observations.query_criteria(
+        coordinates=coords,
+        radius=search_radius_arcsec * u.arcsec,
+        obs_collection="GALEX"
+    )
+
+    if len(obs_table) == 0:
+        print("No GALEX observations found for the target.")
+        return []
+
+    if verbose:
+        print(f"Found {len(obs_table)} observations. Getting product lists...")
+
+    product_tables = [Observations.get_product_list(obs) for obs in obs_table]
+    all_products = vstack(product_tables)
+
+
+    # Filter for SCIENCE + skybg files
+    is_sci = (all_products['productType'] == 'SCIENCE') & (all_products['dataproduct_type'] == 'image')
+    desc_col = all_products['description']
+    desc_filled = np.array([str(x) if x is not np.ma.masked else "" for x in desc_col])
+    is_bkg = np.char.find(desc_filled, 'skybg') >= 0
+    product_subset = all_products[is_sci | is_bkg]
+
+        # Count and list how many match
+    print(f"Found {np.sum(is_bkg)} files with 'skybg' in the filename")
```

Comment:
put this behind `verbose`, too. In practice all prints should be either in instead a logger or behind verbose.